ambry-api/src/test/java/com.github.ambry/rest/RequestPathTest.java

```diff
@@ -106,11 +123,32 @@ public void testOperationMatching() {
    */
   @Test
   public void testStripLeadingSlash() {
-    RequestPath requestPath = new RequestPath("", "", "", "/opToMatch/abc", null);
+    RequestPath requestPath = new RequestPath("", "", "", "/opToMatch/abc", null, NO_BLOB_SEGMENT_IDX_SPECIFIED);
     assertEquals("Should strip leading slash", "opToMatch/abc", requestPath.getOperationOrBlobId(true));
     assertEquals("Should not strip leading slash", "/opToMatch/abc", requestPath.getOperationOrBlobId(false));
   }
 
+  /**
+   * Tests that the expected exception is thrown when a Segment request comes in without a valid
+   * integer index
+   * @throws UnsupportedEncodingException
+   * @throws URISyntaxException
+   */
+  @Test
+  public void testBadSegmentSubResourceNumber() throws UnsupportedEncodingException, URISyntaxException {
+    String requestPath = "/mediaexpectedOpOrId/Segment/notAnInteger";
+    List<String> prefixesToRemove = new ArrayList<>();
+    RestRequest restRequest = RestUtilsTest.createRestRequest(RestMethod.GET, requestPath, null);
+    try {
+      RequestPath.parse(restRequest, prefixesToRemove, CLUSTER_NAME);
+    } catch (RestServiceException e) {
+      assertTrue(e.getErrorCode().equals(RestServiceErrorCode.BadRequest));
+      //expected
+      return;
+    }
+    fail();
```

Comment:
minor: move `fail()` to line 144 and remove `return` at line 147

Reply:
good suggestion